packages/components/src/components/Select/Combobox.tsx

```diff
@@ -52,77 +57,105 @@ export const SelectCombobox = (
         value,
         defaultValue,
         placeholder = '',
+        allowCustomValue = false,
         disabled,
         'aria-label': ariaLabel,
         'data-test-id': dataTestId = 'fondue-select-combobox',
     }: ComboboxProps,
     forwardedRef: ForwardedRef<HTMLDivElement>,
 ) => {
-    const { inputSlots, menuSlots, items, filterText, clearButton, getItemByValue, setFilterText } =
-        useSelectData(children);
-
-    const defaultItem = getItemByValue(defaultValue);
-    const activeItem = getItemByValue(value);
+    const {
+        inputSlots,
+        menuSlots,
+        items,
+        filterText,
+        clearButton,
+        valueExists,
+        existingValueItem,
+        valueInvalid,
+        getItemByValue,
+        setFilterText,
+    } = useSelectData(children, allowCustomValue);
 
     const {
+        isOpen,
+        highlightedIndex,
+        inputValue,
+        selectedItem,
+        selectItem,
         getInputProps,
         getToggleButtonProps,
         getMenuProps,
         getItemProps,
         reset,
-        isOpen,
-        highlightedIndex,
-        inputValue,
     } = useCombobox({
         items,
+        defaultHighlightedIndex: 0,
+        selectedItem: getItemByValue(value),
+        defaultSelectedItem: getItemByValue(defaultValue),
         onSelectedItemChange: ({ selectedItem }) => {
             onSelect && onSelect(selectedItem.value);
         },
-        selectedItem: activeItem,
         onInputValueChange: ({ inputValue }) => {
             setFilterText(inputValue);
         },
-        defaultSelectedItem: defaultItem,
-        defaultHighlightedIndex: 0,
         itemToString: (item) => (item ? item.label : ''),
     });
 
     const wasClicked = useRef(false);
-    const valueInvalid = useMemo(
-        () => inputValue && !items.find((item) => item.label.toLowerCase().includes(inputValue.toLowerCase())),
-        [inputValue, items],
-    );
+
+    const onBlurHandler = (blurEvent: FocusEvent<HTMLInputElement, Element>) => {
+        blurEvent.target.dataset.showFocusRing = 'false';
+        wasClicked.current = false;
+
+        if (valueExists) {
+            if (existingValueItem?.label) {
```

Comment:
out of scope, this doesn't happen as of now, unrelated from the custom value

if we change this, please create a separete pr

Reply:
I was about to copy the code from this over to a new branch when it occurred to me this will cause conflicts down the line if I implement it separately. For now, it's just a suggestion, we might get rid of it entirely but if we do decide to keep it, doesn't it make sense that it would be included alongside a new feature that makes it necessary?

Reply:
This "select on blur" behaviour is unrelated to the custom value change.

This only changes the behaviour when choosing a option thats pre existing. 
There is no change on this behaviour when `allowCustomOptions` is passed or not.

We're basically bundling two interaction changes together. 

Let's look at it together later